File: Application/src/main/java/Contracts/IOrderService.java
package Contracts;

import Entities.Order;

import java.util.ArrayList;

public interface IOrderService {
    public ArrayList<Order> GetAllOrders();
    public Order GetOrderById(int id);
    public void AddOrder(Order order); //TODO
    public void DeleteOrderByIdAsync(int id);
    public void UpdateOrderAsync(Order order);
}

File: Persistence/Data/Contracts/IOrderService.cs
using Entities;

namespace Contracts;

public interface IOrderService
{
    public Task<IList<Order>> GetAllOrdersAsync();
    public Task<Order> GetOrderById(int id);
    public Task AddOrderAsync(Order order);
    public Task DeleteOrderByIdAsync(int id);
    public Task UpdateOrderAsync(Order order);
}